lib/util.h

```diff
@@ -94,20 +94,27 @@ extern const char CYRUS_VERSION[];
 #endif
 
 #define BIT32_MAX 4294967295U
-
-#if UINT_MAX == BIT32_MAX
-typedef unsigned int bit32;
-#elif ULONG_MAX == BIT32_MAX
-typedef unsigned long bit32;
-#elif USHRT_MAX == BIT32_MAX
-typedef unsigned short bit32;
+#define BIT64_MAX 18446744073709551615UL
+
+#if ULONG_MAX == BIT64_MAX
+#define BIT64_FMT          "%016lx"
+#define UINT64_FMT         "%lu"
+#define UINT64_LALIGN_FMT  "%-*lu"
+#define UINT64_NANOSEC_FMT ".%.9lu"
+#elif ULLONG_MAX == BIT64_MAX
+#define BIT64_FMT          "%016llx"
+#define UINT64_FMT         "%llu"
+#define UINT64_LALIGN_FMT  "%-*llu"
+#define UINT64_NANOSEC_FMT ".%.9llu"
 #else
-#error dont know what to use for bit32
+#error dont know what to use for bit64
 #endif
```

Comment:
(@dilyanpalauzov mentioned this also)

You should be able to remove the conditionalisation here by just defining these in terms of PRIu64 et al, which are defined in `<inttypes.h>`.  Something like:

```
#define BIT64_FMT "%016" PRIx64 /* n.b. x rather than u */
#define UINT64_FMT "%" PRIu64 /* XXX this is sorta redundant though */
#define UINT64_LALIGN_FMT "%-*" PRIu64
#define UINT64_NANOSEC_FMT ".%.9" PRIu64
```

Reply:
Done on 64bit-nanosecond-jmapids-bis branch and then rebase this fork